populator-machinery/controller.go

```diff
@@ -515,13 +523,21 @@ func (c *controller) runWorker() {
 		}
 		var err error
 		parts := strings.Split(key, "/")
+		var pvc *corev1.PersistentVolumeClaim
 		switch parts[0] {
 		case "pvc":
 			if len(parts) != 3 {
 				utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
 				return nil
 			}
-			err = c.syncPvc(context.TODO(), key, parts[1], parts[2])
+			pvcNamespace := parts[1]
+			pvcName := parts[2]
+			pvc, err = c.pvcLister.PersistentVolumeClaims(pvcNamespace).Get(pvcName)
```

Comment:
why we need this?

Reply:
We need the PVC UID, which is not available until we get into syncPVC.

Reply:
can you try if add to key works? https://github.com/kubernetes-csi/lib-volume-populator/blob/release-2.0/populator-machinery/controller.go#L327 so we can parse the uid from the key